src/graph/matrix/ffi.rs

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright FalkorDB Ltd. 2023 - present
+ * Licensed under the Server Side Public License v1 (SSPLv1).
+ */
+
+use std::ptr::null_mut;
+
+use super::{
+    delta_matrix::DeltaMatrix,
+    delta_matrix_iter::DeltaMatrixIter,
+    GraphBLAS::{GrB_Index, GrB_Info, GrB_Matrix, GrB_Semiring, GrB_Type, GrB_Vector},
+};
+
+type _Matrix = *mut DeltaMatrix;
+type _MatrixTupleIter = *mut DeltaMatrixIter<'static>;
+
+#[no_mangle]
+unsafe extern "C" fn Delta_Matrix_new(
+    a: *mut _Matrix,
+    ty: GrB_Type,
+    nrows: GrB_Index,
+    ncols: GrB_Index,
+    transpose: bool,
+) -> GrB_Info {
+    *a = Box::into_raw(Box::new(DeltaMatrix::new(ty, nrows, ncols, transpose)));
+    GrB_Info::GrB_SUCCESS
+}
+
+#[no_mangle]
+unsafe extern "C" fn Delta_Matrix_getTranspose(c: _Matrix) -> _Matrix {
+    match (&mut *c).transposed() {
```

Comment:
When calling a function( for example, c.transposed() ), the compiler will automatically convert the dereference into the correct reference, so this is not needed.
I.e.,
(*c).transposed() is the same as (&mut *c).transposed()
because the transposed() function takes a &mut self

Reply:
it is not